### DISCUSSION TITLE
Get it to stop stripping newlines?

### DISCUSSION THREAD

**Discussion by amnuts** [pterm/pterm#694] [Q&A] [answered]
Hey there!

I have a couple questions, but I'm pretty new to Go, and very new to pterm (just started to look into it this evening), so my apologies if these are something very obvious and I'm just missing it in the examples and docs...

I'm trying to output some text with newlines in it.  At the moment, I have:

```go
p := pterm.DefaultParagraph.WithWriter(cmd.OutOrStdout()).WithMaxWidth(90)
b := pterm.DefaultBasicText.WithStyle(pterm.NewStyle(pterm.Bold))
if len(cmd.Short) > 0 {
    p.Println(cmd.Short)
    if len(cmd.Long) > 0 {
        p.Printfln("\n%s", cmd.Long)
    }
} else if len(cmd.Long) > 0 {
    p.Printf(b.Sprint(cmd.Long))
}
```

But the newline in `p.Printfln("\n%s", cmd.Long)` is being removed or not printed out.

If I use an additional `p.Println()` like:

```go
    if len(cmd.Long) > 0 {
        p.Println()
        p.Println(cmd.Long)
    }
```

it does what I want, but I don't want to have to put many additional `p.Println()` calls in there; with some text I have to output, it's going to get a bit messy.

Is there any way to tell the `DefaultParagraph` or `DefaultBasicText` to not strip the newline characters, or do I just have to do the additional `p.Printlbn()` calls?

Also, is there a better way to output a bit of bold text in a paragraph with a ax width of 90 characters, or is what I have there (the `p.Printf(b.Sprint(cmd.Long))`) pretty much how it works?

Thanks!

**Comment by MarvinJWendt** [comment, marked as answer]
Hi!

There is currently no way to stop the stripping of newlines in PTerm components. You may create a feature request if you want to :)

Instead of `b := pterm.DefaultBasicText.WithStyle(pterm.NewStyle(pterm.Bold))` you can do `b := pterm.Bold`, the usage will be the same.

BTW: You only need the `ParagraphPrinter` when you want your text to be a bit more responsive. Usually, a simple `Println` is enough.

**Reply by amnuts** [reply]
That's great to know, @MarvinJWendt, thanks!
